invenio_i18n/views.py

```diff
@@ -53,11 +54,15 @@ def get_redirect_target():
             return target
 
 
-@blueprint.route('/<lang_code>/', methods=['POST'])
+@blueprint.route('/',  methods=['POST'])
+@blueprint.route('/<lang_code>/',  methods=['POST'])
 def set_lang(lang_code=None):
     """Set language in session and redirect."""
     # Check if language is available.
-    lang_code = lang_code.lower()
+    if lang_code is not None:
+        lang_code = lang_code.lower()
+    else:
+        lang_code = request.form['lang_code']
     languages = dict(current_app.extensions['invenio-i18n'].get_languages())
     if lang_code not in languages:
         abort(404)
```

Comment:
By default, right now if you are not making a GET request, it is showing a 405 (Method not allowed). Should it show a 400 error instead?